Mathlib/Algebra/Star/Subalgebra.lean

```diff
@@ -792,6 +792,20 @@ protected def codRestrict (f : A →⋆ₐ[R] B) (S : StarSubalgebra R B) (hf :
   toAlgHom := AlgHom.codRestrict f.toAlgHom S.toSubalgebra hf
   map_star' := fun x => Subtype.ext (map_star f x)
 
+@[simp]
+theorem subtype_comp_codRestrict (f : A →⋆ₐ[R] B) (S : StarSubalgebra R B)
+    (hf : ∀ x : A, f x ∈ S) : S.subtype.comp (f.codRestrict S hf) = f :=
+  StarAlgHom.ext fun _ => rfl
+
+@[simp]
+theorem coe_codRestrict (f : A →⋆ₐ[R] B) (S : StarSubalgebra R B) (hf : ∀ x, f x ∈ S) (x : A) :
```

Comment:
```suggestion
theorem codRestrict_apply (f : A →⋆ₐ[R] B) (S : StarSubalgebra R B) (hf : ∀ x, f x ∈ S) (x : A) :
```
since this mentions `x` (this also matches `LinearMap.codRestrict_apply`)

Reply:
I think it's more important that it mirrors [AlgHom.coe_codRestrict](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Algebra/Subalgebra/Basic.html#AlgHom.coe_codRestrict) and [NonUnitalStarAlgHom.coe_codRestrict](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Star/NonUnitalSubalgebra.html#NonUnitalStarAlgHom.coe_codRestrict). At some point we want to have a full look at the whole library and make sure that everything is consistent.
I'm going to open issue about that, but I think I prefer to keep this name for now if you think that's okay.

Reply:
#6679

Reply:
Ah, actually I guess these should be called `coe_codRestrict_apply`, since the coercion is `Subtype.val` not just `FunLike.coe`. But yes, I'm fine leaving the name as is in light of my comment below.